testhelpers/flags.go

```diff
@@ -0,0 +1,33 @@
+package testhelpers
+
+import (
+	"flag"
+	"testing"
+)
+
+var integrationTest = flag.Bool("integration", false, "Run the integration go tests")
+var unitTest = flag.Bool("unit", false, "Run the unit go tests")
+
+var concurrent = flag.Bool("concurrent", true, "Run the go tests concurrently")
```

Comment:
Note to Reviewer(s):

All `t.Parallel()` calls were removed from tests. The call is now made inside functions that determines whether or not a test is ran (currently `UnitTest` or `IntegrationTest`), this means that sub tests (`t.Run()`) will be ran sequentially. It is my _opinion_ that this makes the testing code easier to read, feel free to push back.

The `concurrent` (avoid collision with `test.parallel`) flag has been added with a default value of `true`. This means tests are ran in parallel by default -- the original behaviour. This flag allows us to configure certain categories of tests to run in parallel or serially which I believe could prove useful in debugging test performance down the road, again feel free to push back on this.